sustainability/src/main/webapp/WEB-INF/views/search/results.jsp

```diff
@@ -13,9 +13,49 @@
 	</small>
 </h1>
 
-
 <ul class="list-group">
-	<c:forEach items="${results}" var="entry">
-		<li class="list-group-item"><a href="<c:url value="/text?uri=${entry.key}" />">${entry.key}</a></li>
+	<c:forEach items="${resource}" var="entry" varStatus="vs">
+		<c:if test="${not empty entry and empty entry.updatedOn}">
+			<li id="part${vs.index}" class="list-group-item">
+				<i class="fa fa-spinner fa-pulse"></i> 
+				Please wait while we are loading the text for you.
+			</li>
+			<script>
+				setTimeout(reload('${entry.uri}','${vs.index}'), 3000);
+				function reload(uri, index) {
+				    $.ajax({
+					  url: "<c:url value="/perspective/researcher/resource"/>",
+					  type : 'GET',
+					  data: { 'uri' : uri },
+					  success: function (rsrc) {
+					  	setTimeout(function() {
+							if(rsrc.updatedOn != null) {
+							  var updateResults = "<li class = 'list-group-item'>" +
+								"<a href = /text?uri=" + rsrc.uri + ">" + rsrc.title +
+										" (" +	rsrc.year + ")</a></li>";
+							$('#part' + index).replaceWith(updateResults);
```

Comment:
why is updating of the resource has been loaded delayed? It can be displayed right away.